### PR TITLE
[Doc] Improve the wording of CRD recommendations to avoid misleading users

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

Improve the doc based on @hongchaodeng's recommendation.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
I'll defer to @angelinalg on the best wording here.

File: doc/source/cluster/kubernetes/getting-started.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ It offers 3 custom resource definitions (CRDs):
 
 ## Which CRD should you choose?
 
-Using [RayService](kuberay-rayservice-quickstart) to serve models and using [RayCluster](kuberay-raycluster-quickstart) to develop Ray applications are no-brainer recommendations from us.
-However, if the use case is not model serving or prototyping, how do you choose between [RayCluster](kuberay-raycluster-quickstart) and [RayJob](kuberay-rayjob-quickstart)?
+Use [RayService](kuberay-rayservice-quickstart) to serve models and [RayCluster](kuberay-raycluster-quickstart) to prototype Ray applications as recommended practices.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started.md:18]:
```suggestion
Use [RayService](kuberay-rayservice-quickstart) to serve models.  

Use [RayCluster](kuberay-raycluster-quickstart) to prototype Ray applications.

If you aren't serving models or prototyping an application, the CRD recommendation depends on the answers to the following questions.
```

**Review by angelinalg — APPROVED** [review]

**Comment by kevin85421** [comment]
Thank @angelinalg for the review! I chatted with @hongchaodeng offline, and we decided to not only improve the wording but also add more details about why users should use RayService for model serving and RayCluster for prototyping. Hence, I convert this PR back to draft.

**Comment by angelinalg** [comment]
I'd really like to suggest some changes to the titles of the questions in this section. Will you be touching those? If not, I can open a new PR to address that separately.

**Comment by kevin85421** [comment]
> Will you be touching those? 

No

> If not, I can open a new PR to address that separately.

It will be great!

**Pull request closed without merging**